src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -57,9 +64,21 @@ static PSVersionInfo()
         {
             s_psVersionTable = new PSVersionHashTable(StringComparer.OrdinalIgnoreCase);
 
+            // Get '6.0.0-beta.7 Commits: 29 SHA: 52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.
+            // The formatted string is displayed to users in '$PSVersionTable' and PowerShell banner.
+            string assemblyPath = typeof(PSVersionInfo).Assembly.Location;
+            string formattedGitCommitId = FileVersionInfo.GetVersionInfo(assemblyPath).ProductVersion;
+            s_psVersionTable[PSFormattedGitCommitIdName] = formattedGitCommitId;
+
+            // Convert to '6.0.0-beta.7-29-52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.
+            // The raw git commit version string is widely used internally.
+            s_rawGitCommitId = "v" + formattedGitCommitId.Replace(" Commits: ", "-").Replace(" SHA: ", "-");
```

Comment:
There is a `g` before the SHA value when got from `git describe`:
`v6.0.0-beta.7-30-gf1ec922fd93cf833f32d98fcf337a51f2919b6cf`

And also, when `Commit:` is not present, it indicates the build was from a release tag, so SHA shouldn't be put in the GitCommitId in that case, for example, it would be `v6.0.0-beta.7`.

Reply:
Fixed.